euphonic/qpoint_frequencies.py

```diff
@@ -140,6 +152,13 @@ def _calculate_dos(self, dos_bins: Quantity,
         case, mode_widths and mode_weights must be shape (1, n_modes)
         arrays
         """
+        if adaptive_method!='fast' and error is not None:
+            warnings.warn('\n The error argument can only be applied when using the fast adaptive '
+                          'method, so will be ignored.')
+        if mode_widths is None and adaptive_method!='default' or mode_widths is None and error is not None:
```

Comment:
This seems a bit cumbersome especially as it will _not_ warn somebody against a call to `_calculate_dos(mode_widths=None, adaptive_method='default')` (i.e. they have still non-sensically set an option for adaptive_method, but it happens to equal the default.).

I suggest either
- we don't warn in this case, or
- we let the default value = `None`, warn `if mode_widths is None and adaptive_method is not None` and then set it to a default if None. (The default should be mentioned in docstring.)

Reply:
Yeah that's true, I'll change the default value to None and add this to the docstring.